MediaBrowser.Model/Dlna/StreamBuilder.cs

```diff
@@ -386,32 +387,33 @@ private static TranscodeReason GetTranscodeReasonForFailedCondition(ProfileCondi
         /// <param name="type">The <see cref="DlnaProfileType"/>.</param>
         /// <param name="playProfile">The <see cref="DirectPlayProfile"/> object to get the video stream from.</param>
         /// <returns>The normalized input container.</returns>
-        public static string? NormalizeMediaSourceFormatIntoSingleContainer(string inputContainer, DeviceProfile? profile, DlnaProfileType type, DirectPlayProfile? playProfile = null)
+        public static string NormalizeMediaSourceFormatIntoSingleContainer(string inputContainer, DeviceProfile? profile, DlnaProfileType type, DirectPlayProfile? playProfile = null)
         {
-            if (string.IsNullOrEmpty(inputContainer))
+            if (!inputContainer.Contains(',', StringComparison.OrdinalIgnoreCase) || profile is null)
             {
-                return null;
+                return inputContainer;
             }
 
-            var formats = ContainerProfile.SplitValue(inputContainer);
-
-            if (profile is not null)
+            var formats = ContainerHelper.Split(inputContainer);
```

Comment:
I thought you wanted to minimize the allocation by using Span.
If you are going to leave it as is, remove `ToString` below.

Reply:
If I understood it correctly, we can't use Span if we want to remove empty. The current SplitSpan implementation just converts the string to a Span Enumerator.
This PR was initially only a forward-port of an already existing one 😅

Reply:
Oh, I thought that was a built-in feature. nvm then.

For further optimization (separate PR), we need to improve this by adding split flags:
https://github.com/jellyfin/jellyfin/blob/751e12e5b5c59f9df4494251166f47c8cf9ebfe1/src/Jellyfin.Extensions/SplitStringExtensions.cs#L93